grammar.js

```diff
@@ -193,9 +233,11 @@ module.exports = grammar(require('tree-sitter-javascript/grammar'), {
       seq($.destructuring_pattern, optional($.type_annotation), $._initializer)
     ),
 
-    ambient_method_declaration: $ => seq(
+    abstract_method_definition: $ => seq(
```

Comment:
We currently don't define an `ambient_class`, and the way the existing `ambient_method_declaration` was written caused a lot of issues for parsing method definitions without statement blocks. I think this change improves the grammar as a whole to make ambient method declarations more consistent with how we currently model ambient class definitions. It also I think helps us define and name the notion of `abstract` methods and classes now, rather than keeping that semantic detail hidden in the tree.